plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeTranslator.java

```diff
@@ -171,6 +173,10 @@ private OrcTypeTranslator() {}
                     DecimalType.createDecimalType(fromOrcType.getPrecision().orElseThrow(), fromOrcType.getScale().orElseThrow()),
                     varcharType));
         }
+
+        if (fromOrcTypeKind == BINARY && toTrinoType instanceof VarcharType varcharType) {
+            return Optional.of(createVarbinaryToVarcharCoercer(varcharType, true));
+        }
```

Comment:
Are you sure this is needed ? My impression is that this should already work in orc reader

Reply:
ORC coercion for varbinary to varchar is quite different - it's coerces to the hex representation directly - unlike other file format